contracts/NFTStorefrontV2.cdc

```diff
@@ -182,11 +182,16 @@ pub contract NFTStorefrontV2 {
         ) {
 
             pre {
-                // Validate the expiry
-                expiry > UInt64(getCurrentBlock().timestamp) : "Expiry should be in the future"
                 // Validate the length of the sale cut
                 saleCuts.length > 0: "Listing must have at least one payment cut recipient"
             }
+            assert (
+                // Validate the expiry
+                expiry > UInt64(getCurrentBlock().timestamp),
```

Comment:
Did this get changed from a pre-condition to an assert because getCurrentBlock is maybe not a view function?

Reply:
Correct, I was kind of confused why it isn't a pure function. I asked in the #cadence channel in discord

Reply:
View functions are a Stable Cadence feature, i.e. it is not released yet, so developers on Discord will not be able to assist with this.

When introducing the feature we had to annotate all existing functions with the view modifier where applicable and seem to have missed the block functions. I'll make them view and we can keep the pre-conditions.

Reply:
I'm just trying to ask more of my questions in public. Stable cadence is still public information, so I figured it might be useful to ask it there